### PR TITLE
Process actions sync. on startup; don't dupe nonexistent profile

### LINKED ISSUE #2455: Bug Report: Cannot find profile --> Crash
# Environment

Introduced by #2422 

Since we'll be defaulting to the Windows Terminal default settings, this means that the user may be working with profiles that don't exist AFTER the proper settings are imported. Below is an example of such a case, but several other places may cause the same issue to arise.

I'd bet that this is because we usually use "FindProfile()" (or something like that) and we just don't handle that properly.

# Steps to reproduce

- create a tab from the WT default settings
- import your custom settings
- duplicate the created tab

# Expected behavior

Not sure. Maybe a dialogue box saying "This operation could not be completed"?

# Actual behavior

CRASH!

### LINKED ISSUE #4618: wt command-line cannot consistently parse more than one argument
Bottom line: the more arguments I put on `wt` the less likely it is to work. This is currently unusable even for launching the app clean.

Something like this, in a run dialog:
```
wt new-tab -p "PowerShell Core"; split-pane -H -p "Windows PowerShell"; split-pane -p cmd
```

Seems to work less than one time in five on my computer. Most of the time I get two panes, not three, and occasionally only one.

# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18363.657]
Windows Terminal Version: 0.9.433.0
```

# Steps to reproduce

Run the command above from a run dialog a half dozen times

# Expected behavior

The window should open the same way every time, with three panes, one for each of the shells I have in Windows...

# Actual behavior

Sometimes just PowerShell Core, sometimes both PowerShells, rarely all three including cmd.

### PR DESCRIPTION
**Pull request by zadjii-msft**
This PR has evolved to encapsulate two related fixes that I can't really
untie anymore.

#2455 - Duplicating a tab that doesn't exist anymore

This was the bug I was originally fixing in #4429. 

When the user tries to `duplicateTab` with a profile that doesn't exist
anymore (like might happen after a settings reload), don't crash.

As I was going about adding tests for this, got blocked by the fact that
the Terminal couldn't open _any_ panes while the `TerminalPage` was size
0x0. This had two theoretical solutions: 

* Fake the `TerminalPage` into thinking it had a real size in the test -
  probably possible, though I'm unsure how it would work in practice.
* Change `Pane`s to not require an `ActualWidth`, `ActualHeight` on
  initialization. 

Fortuately, the second option was something else that was already on my
backlog of bugs. 

#4618 - `wt` command-line can't consistently parse more than one arg

Presently, the Terminal just arbitrarily dispatches a bunch of handlers
to try and handle all the commands provided on the commandline. That's
lead to a bunch of reports that not all the commands will always get
executed, nor will they all get executed in the same order. 

This PR also changes the `TerminalPage` to be able to dispatch all the
commands sequentially, all at once in the startup. No longer will there
be a hot second where the commands seem to execute themselves in from of
the user - they'll all happen behind the scenes on startup. 

This involved a couple other changes areound the `TerminalPage`
* I had to make sure that panes could be opened at a 0x0 size. Now they
  use a star sizing based off the percentage of the parent they're
  supposed to consume, so that when the parent _does_ get laid out,
  they'll take the appropriate size of that parent.
* I had to do some math ahead of time to try and calculate what a
  `SplitState::Automatic` would be evaluated as, despite the fact that
  we don't actually know how big the pane will be. 
* I had to ensure that `focus-tab` commands appropriately mark a single
  tab as focused while we're in startup, without roundtripping to the
  Dispatcher thread and back

## References

#4429 - the original PR for #2455
#5047 - a follow-up task from discussion in #4429
#4953 - a PR for making panes use star sizing, which was immensly
        helpful for this PR.

## Detailed Description of the Pull Request / Additional comments

`CascadiaSettings::BuildSettings` can throw if the GUID doesn't exist.
This wraps those calls up with a try/catch.

It also adds a couple tests - a few `SettingsTests` for try/catching
this state. It also adds a XAML-y test in `TabTests` that creates a
`TerminalPage` and then performs som UI-like actions on it. This test
required a minor change to how we generate the new tab dropdown - in the
tests, `Application::Current()` is _not_ a `TerminalApp::App`, so it
doesn't have a `Logic()` to query. So wrap that in a try/catch as well.

While working on these tests, I found that we'd crash pretty agressively
for mysterious reasons if the TestHostApp became focused while the test
was running. This was due to a call in
`TSFInputControl::NotifyFocusEnter` that would callback to
`TSFInputControl::_layoutRequested`, which would crash on setting the
`MaxSize` of the canvas to a negative value. This PR includes a hotfix
for that bug as well. 

## Validation Steps Performed

* Manual testing with a _lot_ of commands in a commandline
* run the tests
* Team tested in selfhost

Closes #2455
Closes #4618

### CODE REVIEW

File: src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp [hunk 1]
```diff
@@ -2094,6 +2098,146 @@ namespace TerminalAppLocalTests
...
+        const auto settingsJsonObj = VerifyParseSucceeded(settingsString);
+        auto settings = CascadiaSettings::FromJson(settingsJsonObj);
+        settings->_ValidateSettings();
+
+        VERIFY_ARE_EQUAL(2u, settings->_warnings.size());
+        VERIFY_ARE_EQUAL(2u, settings->_profiles.size());
+        VERIFY_ARE_EQUAL(settings->_globals.GetDefaultProfile(), settings->_profiles.at(0).GetGuid());
+        try
+        {
+            const auto [guid, termSettings] = settings->BuildSettings(nullptr);
```
Comment (DHowett-MSFT) [inline src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp]: this just looks like another test of building w/ a null guid; what's the link between the default profile guid and the nullptr here?

File: src/cascadia/LocalTests_TerminalApp/TabTests.cpp [hunk 2]
```diff
@@ -164,4 +176,325 @@ namespace TerminalAppLocalTests
...
+    //     the Application will make it behave as expected.
+    //   * It will wait for the TerminalPage to finish initialization before
+    //     returning control to the caler. It does this by creating an event and
+    //     only setting the event when the TerminalPage raises its Initialized
+    //     event, to signal that startup is complete. At this point, there will
+    //     be one tab with the default profile in the page.
+    //   * It will also ensure that the first tab is focused, since that happens
+    //     asynchronously in the application typically.
+    // Arguments:
+    // - projectedPage: a TerminalPage projected type that will recieve the new TerminalPage instance
```
Comment (DHowett-MSFT) [inline src/cascadia/LocalTests_TerminalApp/TabTests.cpp]: projected type can be trivially generated from the com_ptr one; do we need both?

File: src/cascadia/TerminalApp/Pane.cpp [hunk 3]
```diff
@@ -793,20 +793,24 @@ void Pane::_SetupChildCloseHandlers()
...
+        // const auto paneSizes = _CalcChildrenSizes(rootSize.Width);
 
         auto firstColDef = Controls::ColumnDefinition();
-        firstColDef.Width(GridLengthHelper::FromValueAndType(paneSizes.first, GridUnitType::Star));
+        // firstColDef.Width(GridLengthHelper::FromValueAndType(paneSizes.first, GridUnitType::Star));
+        firstColDef.Width(GridLengthHelper::FromValueAndType(first, GridUnitType::Star));
 
         auto secondColDef = Controls::ColumnDefinition();
-        secondColDef.Width(GridLengthHelper::FromValueAndType(paneSizes.second, GridUnitType::Star));
+        // secondColDef.Width(GridLengthHelper::FromValueAndType(paneSizes.second, GridUnitType::Star));
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Pane.cpp]: comments

File: src/cascadia/TerminalApp/Pane.cpp [hunk 4]
```diff
@@ -991,6 +1007,20 @@ SplitState Pane::_convertAutomaticSplitState(const SplitState& splitType) const
...
+    if (splitType == SplitState::Automatic && _isSizeZero())
+    {
+        return _splitState == SplitState::Horizontal ? SplitState::Vertical : SplitState::Horizontal;
+    }
+    return splitType;
+}
+
+bool Pane::_isSizeZero() const
+{
+    return _root.ActualWidth() == 0 && _root.ActualHeight() == 0;
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Pane.cpp]: Wonder if it's safer to use `||` here?

File: src/cascadia/TerminalApp/Pane.cpp [hunk 5]
```diff
@@ -991,6 +1007,20 @@ SplitState Pane::_convertAutomaticSplitState(const SplitState& splitType) const
     return splitType;
 }
 
+SplitState Pane::_preTranslateAutoSplitType(const SplitState& splitType) const
+{
+    if (splitType == SplitState::Automatic && _isSizeZero())
+    {
+        return _splitState == SplitState::Horizontal ? SplitState::Vertical : SplitState::Horizontal;
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Pane.cpp]: this deserves an explanatory comment!

File: src/cascadia/TerminalApp/Pane.cpp [hunk 6]
```diff
@@ -1002,6 +1032,9 @@ bool Pane::_CanSplit(SplitState splitType)
     const Size actualSize{ gsl::narrow_cast<float>(_root.ActualWidth()),
                            gsl::narrow_cast<float>(_root.ActualHeight()) };
 
+    const Size desiredSize = _root.DesiredSize();
+    desiredSize;
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Pane.cpp]: I don't like this at all: are we using a getter for a side effect?

If so, `(void)_this.Thing()` with a comment explaining it

File: src/cascadia/TerminalApp/Tab.cpp [hunk 7]
```diff
@@ -406,19 +426,31 @@ namespace winrt::TerminalApp::implementation
...
+    // - See Pane::PreCalculateAutoSplit
+    // Arguments:
+    // - availableSpace: The theoretical space that's available for this Tab's content
+    // Return Value:
+    // - The SplitState that we should use for an `Automatic` split given
+    //   `availableSpace`
+    SplitState Tab::PreCalculateAutoSplit(winrt::Windows::Foundation::Size availableSpace) const
+    {
+        const auto result = _rootPane->PreCalculateAutoSplit(_activePane, availableSpace);
+        return result.has_value() ? result.value() : SplitState::Vertical;
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Tab.cpp]: this is a long way to say `result.value_or(SplitState::Vertical)` :smile:

File: src/cascadia/TerminalApp/Tab.h [hunk 8]
```diff
@@ -20,6 +26,7 @@ namespace winrt::TerminalApp::implementation
         winrt::Windows::UI::Xaml::UIElement GetRootElement();
         winrt::Microsoft::Terminal::TerminalControl::TermControl GetActiveTerminalControl() const;
         std::optional<GUID> GetFocusedProfile() const noexcept;
+        // std::weak_ptr<Pane> GetActivePane() const noexcept;
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/Tab.h]: dead

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 9]
```diff
@@ -2207,6 +2211,8 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         // If we haven't initialized yet, just quick return.
         if (!_terminal)
         {
+            // fake it
+            eventArgs.CurrentPosition({ 0, 0 });
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalControl/TermControl.cpp]: this will conflict with #5051

File: src/cascadia/TerminalControl/TSFInputControl.cpp [hunk 10]
```diff
@@ -172,8 +172,10 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         TextBlock().FontSize(fontSizePx);
         TextBlock().FontFamily(Media::FontFamily(fontArgs->FontFace()));
 
-        const auto widthToTerminalEnd = Canvas().ActualWidth() - ::base::ClampedNumeric<double>(clientCursorPos.X);
-        TextBlock().MaxWidth(widthToTerminalEnd);
+        const auto canvasActualWidth = Canvas().ActualWidth();
+        const auto widthToTerminalEnd = canvasActualWidth - ::base::ClampedNumeric<double>(clientCursorPos.X);
+        const auto newMaxWidth = std::max<double>(0.0, widthToTerminalEnd);
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalControl/TSFInputControl.cpp]: this seems like it's from Leon's PR

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 11]
```diff
@@ -63,11 +63,20 @@ namespace winrt::TerminalApp::implementation
         _tabView = _tabRow.TabView();
         _rearranging = false;
 
-        // GH#3581 - There's a platform limitation that causes us to crash when we rearrange tabs.
-        // Xaml tries to send a drag visual (to wit: a screenshot) to the drag hosting process,
-        // but that process is running at a different IL than us.
-        // For now, we're disabling elevated drag.
-        const auto isElevated = ::winrt::Windows::UI::Xaml::Application::Current().as<::winrt::TerminalApp::App>().Logic().IsElevated();
+        // GH#2455 - Make sure to try/catch calls to Application::Current,
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: ah, sorry :smile:

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 12]
```diff
@@ -137,51 +146,79 @@ namespace winrt::TerminalApp::implementation
...
+        // Once the page is actually laid out on the screen, trigger all our
+        // startup actions. Things like Panes need to know at least how big the
+        // window will be, so they can subdivide that space.
+        //
+        // _OnFirstLayout will remove this handler so it doesn't get called more than once.
+        _layoutUpdatedRevoker = _tabContent.LayoutUpdated(winrt::auto_revoke, { this, &TerminalPage::_OnFirstLayout });
+    }
+
+    // Method Description:
+    // - This method is caled once on startup, on the first LayoutUpdated event.
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]:
```suggestion
    // - This method is called once on startup, on the first LayoutUpdated event.
```

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 13]
```diff
@@ -137,51 +146,79 @@ namespace winrt::TerminalApp::implementation
...
         }
-
-        // Get the next action to be processed
-        auto nextAction = _appArgs.GetStartupActions().front();
-        _appArgs.GetStartupActions().pop_front();
-
         auto weakThis{ get_weak() };
 
         // Handle it on the UI thread.
         co_await winrt::resume_foreground(Dispatcher(), CoreDispatcherPriority::Low);
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: We may want to get onto the normal priority queue _or_ make this no longer a coroutine since LayoutUpdated comes in on the dispatcher queue already

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: if we get if off the queue entirely, we can dispense with all of the weak stuff _and_ the `_appArgs` check (because iterating over an empty list will be a noop)

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 14]
```diff
@@ -477,27 +522,36 @@ namespace winrt::TerminalApp::implementation
...
+                TraceLoggingUInt32(tabCount, "TabCount", "Count of tabs currently opened in TerminalApp"),
+                TraceLoggingBool(usedManualProfile, "ProfileSpecified", "Whether the new tab specified a profile explicitly"),
+                TraceLoggingGuid(profileGuid, "ProfileGuid", "The GUID of the profile spawned in the new tab"),
+                TraceLoggingBool(settings.UseAcrylic(), "UseAcrylic", "The acrylic preference from the settings"),
+                TraceLoggingFloat64(settings.TintOpacity(), "TintOpacity", "Opacity preference from the settings"),
+                TraceLoggingWideString(settings.FontFace().c_str(), "FontFace", "Font face chosen in the settings"),
+                TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance));
+        }
+        CATCH_LOG();
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: would you mind making this a function-level try? it'll get rid of the extra level of indentation and be in line with what we've been doing lately.

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 15]
```diff
@@ -574,7 +628,8 @@ namespace winrt::TerminalApp::implementation
         if (isFirstTab)
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: random q: shouldn't the tabview handle this now?

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 16]
```diff
@@ -900,20 +964,36 @@ namespace winrt::TerminalApp::implementation
             // Wraparound math. By adding tabCount and then calculating modulo tabCount,
             // we clamp the values to the range [0, tabCount) while still supporting moving
             // leftward from 0 to tabCount - 1.
-            _SetFocusedTabIndex(((tabCount + *index + (bMoveRight ? 1 : -1)) % tabCount));
+            const auto newTabIndex = ((tabCount + *index + (bMoveRight ? 1 : -1)) % tabCount);
+            _SelectTab(newTabIndex);
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: ?

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 17]
```diff
@@ -900,20 +964,36 @@ namespace winrt::TerminalApp::implementation
...
     // true iff we were able to select that tab index, false otherwise
     bool TerminalPage::_SelectTab(const uint32_t tabIndex)
     {
         if (tabIndex >= 0 && tabIndex < _tabs.Size())
         {
-            _SetFocusedTabIndex(tabIndex);
+            if (_startupState == StartupState::InStartup)
+            {
+                auto tab{ _GetStrongTabImpl(tabIndex) };
+                _tabView.SelectedItem(tab->GetTabViewItem());
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: so this is to handle the commandline action to change the focused tab. neat

**Comment by DHowett-MSFT** [comment]
Does this fix the minor visual issue where you do `wt ; ; ; ; ;` and the tabs all pop into existence slowly?

File: src/cascadia/TerminalApp/Pane.cpp (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp]: gah the last commit didn't get pushed ;__;

**Comment by zadjii-msft** [comment]
> 
> 
> Does this fix the minor visual issue where you do `wt ; ; ; ; ;` and the tabs all pop into existence slowly?

Yes, yes it does :)

File: src/cascadia/TerminalApp/Pane.cpp (hunk 6 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp]: Oh no, that's dead code

File: src/cascadia/TerminalApp/Pane.cpp (hunk 4 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp]: This should actually just be gone

File: src/cascadia/TerminalApp/Pane.cpp (hunk 5 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp]: (this is actually dead code)

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 18]
```diff
@@ -1525,14 +1645,26 @@ namespace winrt::TerminalApp::implementation
...
+
+                for (auto tab : _tabs)
+                {
+                    // Attempt to reload the settings of any panes with this profile
+                    auto tabImpl{ _GetStrongTabImpl(tab) };
+                    tabImpl->UpdateSettings(settings, profileGuid);
+                }
             }
+            CATCH_LOG();
+            // TODO: Should we display a dialog when we do nothing because we
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: bare TODO

File: src/cascadia/TerminalControl/TSFInputControl.cpp [hunk 19]
```diff
@@ -172,8 +172,13 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
         TextBlock().FontFamily(Media::FontFamily(fontArgs->FontFace()));
 
-        const auto widthToTerminalEnd = Canvas().ActualWidth() - ::base::ClampedNumeric<double>(clientCursorPos.X);
-        TextBlock().MaxWidth(widthToTerminalEnd);
+        const auto canvasActualWidth = Canvas().ActualWidth();
+        const auto widthToTerminalEnd = canvasActualWidth - ::base::ClampedNumeric<double>(clientCursorPos.X);
+        // Make sure that we're setting the MaxWidth to a positive number - a
+        // negative number here will crash us in mysterious ways with a useless
+        // stack trace
+        const auto newMaxWidth = std::max<double>(0.0, widthToTerminalEnd);
```
Comment (leonMSFT) [inline src/cascadia/TerminalControl/TSFInputControl.cpp]: I'm sorry I don't know why I thought a clamped double would be clamped at 0...

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 11 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: lol np, this wasn't ever tested before so it wasn't a problem 😄

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 16 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Oh, see, `_SetFocusedTabIndex` will only do the dispatcher thing, while `_SelectTab` will to the thing _now_ if we're in startup, otherwise it'll do the `_SetFocusedTabIndex` thing. This basically just makes `focus-tab -n` and `focus-tab -p` work

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 13 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Curiously, it doesn't work if you do it immediately on _this_ pass of the UI thread. I will bump it up to Normal though

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 9 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.cpp]: (I stole it from that PR 😄)

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 15 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Oh yea we don't need this anymore

File: src/cascadia/LocalTests_TerminalApp/TabTests.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/LocalTests_TerminalApp/TabTests.cpp]: Okay yea I thought I might still need that for wackiness's sake. Removed

File: src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp]: `BuildSettings(nullptr)` calls the `BuildSettings(NewTerminalArgs)` version of the method, that will default to the default profile, since no profile was passed in via the `NewTerminalArgs`

**Comment by zadjii-msft** [comment]
Sorry all for submitting the PR before pushing the last code cleanup commit 😅 Things should be much better now

File: src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp (hunk 1 above)
Comment (DHowett-MSFT) [reply src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp]: oh, i see. cool

**Comment by DHowett-MSFT** [comment]
I'm producing a selfhost test build with this pre-merged; it is likely ready, but I'll let you make that call.

**Comment by zadjii-msft** [comment]
Note to the team: I'm giving this like 48 hours of selfhost before we ship it. Hence why I haven't AutoMerge'd this yet ☺️

**Comment by DHowett-MSFT** [comment]
This feels so much better. We should just hashtag shipit

**Comment by DHowett-MSFT** [comment]
@jsoref this may be what you were talking about on the other thread; I’m not getting a green check here despite merging in master?

**Comment by jsoref** [comment]
@DHowett-MSFT : hmm, that isn't the right explanation for this. @german-one 's action is running in his fork (which is good) -- https://github.com/german-one/terminal/actions. The action should also be running on branches in this repo, but it doesn't appear to be: https://github.com/Microsoft/terminal/actions.

Fwiw, roughly the same configuration is in play here: https://github.com/check-spelling/examples-testing/actions and it's working:
https://github.com/check-spelling/examples-testing/commit/0a7e8a2789062362861a517bc25f03fca5fa185c 

Is it possible that these branches also have tags? If so, we might try removing:
https://github.com/microsoft/terminal/blob/master/.github/workflows/spelling.yml#L6-L7

I added that because for our repositories we were experiencing double builds in github.

**Comment by jsoref** [comment]
@DHowett-MSFT : can you try making a branch of https://github.com/jsoref/terminal/pull/new/drop-tags-ignore in Microsoft/terminal? (no PR, just pull that branch into your repo and push it into github with a branch name)

**Comment by DHowett-MSFT** [comment]
https://github.com/microsoft/terminal/commit/6f717ae9ab3c3150055492fc7ba7895e6b69b72b

**Comment by DHowett-MSFT** [comment]
See, this is why I wanted to run spell on this branch :D
